routes/user.js

```diff
@@ -2,100 +2,122 @@
 
 const AppConstants = require("../app-constants");
 
-const crypto = require("crypto");
 const express = require("express");
 const router = express.Router();
 
 const EmailUtils = require("../email-utils");
 const Subscribers = require("../subscribers");
 
-// We send verification emails to addresses that want to subscribe.
-// These addresses are temporarily stored here, mapped to the unique
-// string token that is used for verification.
-const gUnverifiedEmails = new Map();
+const ResponseCodes = Object.freeze({
+  InternalError: 999,
+  EmailNotProvided: 100,
+  EmailNotFound: 101,
+  TokenMismatch: 102,
+});
+
+function resDatabaseError(res, error) {
+  console.log(error);
+  res.status(500).json({
+    error_code: ResponseCodes.InternalError,
+    info: "Database error.",
```

Comment:
Since we're now passing the `error` argument, would there be any value in trying to say what the database error was?
For example:
```js
info: `Database error: ${error.message}`,
```

Reply:
I left this out kind of intentionally. We're logging the error server-side, I don't see any reason to give clients too much information :P 

With the info strings I'm aiming to give clients enough to get back to us with a hint of what error they're seeing.